components/dsp_processor_settings/dsp_processor_settings.c

```diff
@@ -0,0 +1,654 @@
+		ESP_LOGE(TAG, "%s: Failed to acquire mutex (timeout)", __func__);
+		return ESP_ERR_TIMEOUT;
+	}
+
+	nvs_handle_t h;
+	esp_err_t err = nvs_open(NVS_NAMESPACE, NVS_READWRITE, &h);
+	if (err != ESP_OK) {
+		ESP_LOGE(TAG, "%s: Failed to open NVS: %s", __func__,
+				 esp_err_to_name(err));
+		xSemaphoreGive(dsp_settings_mutex);
+		return err;
+	}
+
+	err = nvs_set_i32(h, NVS_KEY_ACTIVE_FLOW, (int32_t)flow);
+	if (err == ESP_OK) {
+		err = nvs_commit(h);
+	}
+
+	nvs_close(h);
+	xSemaphoreGive(dsp_settings_mutex);
+
+	if (err == ESP_OK) {
+		ESP_LOGI(TAG, "%s: Active flow saved: %d", __func__, (int)flow);
+	} else {
+		ESP_LOGE(TAG, "%s: Failed to save active flow: %s", __func__,
+				 esp_err_to_name(err));
+	}
+
+	return err;
+}
+
+esp_err_t dsp_settings_load_active_flow(dspFlows_t *flow) {
+	ESP_LOGD(TAG, "%s: entered", __func__);
+
+	if (!flow)
+		return ESP_ERR_INVALID_ARG;
+	if (!dsp_settings_mutex)
+		return ESP_ERR_INVALID_STATE;
+
+	if (xSemaphoreTake(dsp_settings_mutex, pdMS_TO_TICKS(5000)) != pdTRUE) {
+		return ESP_ERR_TIMEOUT;
+	}
+
+	nvs_handle_t h;
+	esp_err_t err = nvs_open(NVS_NAMESPACE, NVS_READONLY, &h);
+	if (err == ESP_OK) {
+		int32_t v = 0;
+		err = nvs_get_i32(h, NVS_KEY_ACTIVE_FLOW, &v);
+		nvs_close(h);
+		if (err == ESP_OK) {
+			*flow = (dspFlows_t)v;
+			ESP_LOGD(TAG, "%s: Active flow from NVS: %d", __func__, (int)*flow);
+		} else if (err != ESP_ERR_NVS_NOT_FOUND) {
+			ESP_LOGW(TAG, "%s: NVS read error: %s", __func__,
+					 esp_err_to_name(err));
+		}
+	}
+
+	xSemaphoreGive(dsp_settings_mutex);
+	return err;
+}
+
+esp_err_t dsp_settings_save_flow_param(dspFlows_t flow, const char *param_name,
+									   int32_t value) {
+	ESP_LOGD(TAG, "%s: flow=%d param=%s value=%d", __func__, (int)flow,
+			 param_name, (int)value);
+
+	if (!param_name)
+		return ESP_ERR_INVALID_ARG;
+	if (!dsp_settings_mutex)
+		return ESP_ERR_INVALID_STATE;
+
+	char key[32];
+	make_flow_key(key, sizeof(key), flow, param_name);
+
+	if (xSemaphoreTake(dsp_settings_mutex, pdMS_TO_TICKS(5000)) != pdTRUE) {
+		return ESP_ERR_TIMEOUT;
+	}
+
+	nvs_handle_t h;
+	esp_err_t err = nvs_open(NVS_NAMESPACE, NVS_READWRITE, &h);
+	if (err != ESP_OK) {
+		xSemaphoreGive(dsp_settings_mutex);
+		return err;
+	}
+
+	err = nvs_set_i32(h, key, value);
+	if (err == ESP_OK) {
+		err = nvs_commit(h);
+	}
+
+	nvs_close(h);
+	xSemaphoreGive(dsp_settings_mutex);
+
+	if (err == ESP_OK) {
+		ESP_LOGD(TAG, "%s: Saved %s=%d", __func__, key, (int)value);
+	} else {
+		ESP_LOGE(TAG, "%s: Failed to save %s: %s", __func__, key,
+				 esp_err_to_name(err));
+	}
+
+	return err;
+}
+
+esp_err_t dsp_settings_load_flow_param(dspFlows_t flow, const char *param_name,
+									   int32_t *value) {
+	ESP_LOGV(TAG, "%s: flow=%d param=%s", __func__, (int)flow, param_name);
+
+	if (!param_name || !value)
+		return ESP_ERR_INVALID_ARG;
+	if (!dsp_settings_mutex)
+		return ESP_ERR_INVALID_STATE;
+
+	char key[32];
+	make_flow_key(key, sizeof(key), flow, param_name);
```

Comment:
Check for valid flow before calling this then `make_flow_key()` could stay as it is